lib/matplotlib/tests/test_build_grid.py

```diff
@@ -0,0 +1,57 @@
+from matplotlib.testing.decorators import check_figures_equal
+import matplotlib.gridspec as gridspec
+
+
+@check_figures_equal()
+def test_basic(fig_test, fig_ref):
+    x = [["A", "A", "B"], ["C", "D", "B"]]
+    grid_axes = fig_test.build_grid(x)
+
+    for k, ax in grid_axes.items():
+        ax.set_title(k)
+
+    gs = gridspec.GridSpec(2, 3, figure=fig_ref)
+    axA = fig_ref.add_subplot(gs[:1, :2])
+    axA.set_title("A")
+
+    axB = fig_ref.add_subplot(gs[:, 2])
+    axB.set_title("B")
+
+    axC = fig_ref.add_subplot(gs[1, 0])
+    axC.set_title("C")
+
+    axD = fig_ref.add_subplot(gs[1, 1])
+    axD.set_title("D")
+
+
+@check_figures_equal(tol=0.005)
+def test_recursive(fig_test, fig_ref):
+    x = [["A", "B", "B"], ["C", "C", "D"]]
+
+    y = [["F"], [x]]
+
+    grid_axes = fig_test.build_grid(y)
+
+    for k, ax in grid_axes.items():
+        ax.set_title(k)
+
+    gs = gridspec.GridSpec(2, 1, figure=fig_ref)
+    axF = fig_ref.add_subplot(gs[0, 0])
+    axF.set_title("F")
+
+    gs_n = gridspec.GridSpecFromSubplotSpec(2, 3, gs[1, 0])
```

Comment:
suggest `gs[1, 0].subplotspec(2, 3)`....

Reply:
should be `gs[1, 0].subgridspec(...)`